rover_video/src/media_server.cpp

```diff
@@ -0,0 +1,117 @@
+#include "screenshot_server.hpp"
+
+int main(int argc, char* argv[])
+{
+    rclcpp::init(argc, argv);
+
+    RCLCPP_INFO(rclcpp::get_logger("media_server"), "Node initialized.");
+
+    try
+    {
+        rclcpp::spin(std::make_shared<CameraNode>());
+    }
+    catch (const std::exception& e)
+    {
+        RCLCPP_FATAL(rclcpp::get_logger("Dead Node"), "Killing node on exception: %s", e.what());
+    }
+
+    rclcpp::shutdown();
+    return 0;
+}
+
+CameraNode::CameraNode():
+    Node("media_server")
+{
+    _srv_control = this->create_service<rover_msgs::srv::CameraControl>(
+        "/rover/video/media_server",
+        [this](const std::shared_ptr<rover_msgs::srv::CameraControl::Request> request,
+               std::shared_ptr<rover_msgs::srv::CameraControl::Response> response)
+        {
+            this->controlIPCam(*request, *response);
+        });
+
+    _msg_position
+        = this->create_subscription<rover_msgs::msg::GpsPosition>("/rover/gps/position",
+                                                                  1,  // What to put as QoS ?
+                                                                  [this](const rover_msgs::msg::GpsPosition& gps_message)
+                                                                  {
+                                                                      this->callbackPosition(gps_message);
+                                                                  });
+}
+
+void CameraNode::controlIPCam(const rover_msgs::srv::CameraControl::Request& request,
+                              rover_msgs::srv::CameraControl::Response& response)
+{
+    RCLCPP_DEBUG(LOGGER, "Entering the controlIPCam function");
```

Comment:
Traces for function entry and exit are not necessary but not a bad pratice. When doing so please take time to also provide arguments values and more infos about the namespace:
```suggestion 
    RCLCPP_DEBUG(this->get_logger(), "Entry CameraNode::controlIPCam(\nrequest\n\t.start = %i\n\t.name = %s\n\ti.p_address = %s\n\t.metadata = %s\tresponse: Not Set\t)",
     request.start,
     request.name,
     request.p_address,
     request.metadata);
```

Reply:
Honnestly both are fine, but using LOGGER in a scope where you're not inside a class which isn't a Node child would not compile. I started using LOGGER when I started coding with ROS2 but I've grown to not like using it because a logger can be created on the fly like this: `rclcpp::get_logger("Custom Logger Name")`. Using `LOGGER` hides which logger type is used (either rclcpp::get_logger("Custom") or this->get_logger()). Use the one you prefer.